src/main.cpp

```diff
@@ -470,6 +473,10 @@ void UpdateBlockAvailability(NodeId nodeid, const uint256 &hash) {
 }
 
 void MaybeSetPeerAsAnnouncingHeaderAndIDs(const CNodeState* nodestate, CNode* pfrom) {
+    if (nLocalServices & NODE_WITNESS) {
```

Comment:
@rebroad Read BIP 141, 143, 144.

Yes, segregated witness full nodes only download from other witness nodes.
There is no way around that: they require the witness data to be able to
verify blocks, and non-witness nodes cannot provide that.

See #8393 for how to make compact blocks support segregated witness.